File: ts/3110.ts
function scoreOfString(s: string): number {
  let counter = 0;

  for (let i = 0; i < s.length - 1; i++) {
    counter += Math.abs(s.charCodeAt(i) - s.charCodeAt(i + 1));
  }

  return counter;
}

File: go/3110.go
package leetcode

import "math"

func scoreOfString(s string) int {
	acc := 0

	for i := 0; i < len(s)-1; i++ {
		acc += int(math.Abs(float64(s[i]) - float64(s[i+1])))
	}

	return acc
}
